crates/client/rpc/src/lib.rs

```diff
@@ -207,36 +264,42 @@ where
                 continue;
             }
 
-            let block_events = block_events.into_iter().skip(continuation_token);
+            let block_events = block_events.into_iter().skip(continuation_token).collect();
```

Comment:
You are collecting twice, which is a waste we tried to avoid when we originally designed this method

Reply:
removed the collect here and changed the argument type of the filter function